src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/ParameterExpander.java

```diff
@@ -273,52 +373,169 @@ private String expandParameters(String gerritCommand, Run r, TaskListener taskLi
+                return labelValue.getBuildUnstableVoteValue();
+            }
+            else {
+                return config.getVerdictCategory(labelName) != null ? config.getVerdictCategory(labelName).getDefaultBuildUnstableReportingValue() : null;
+            }
+        }
+        else if (res == Result.NOT_BUILT) {
+            if (labelValue != null && labelValue.getBuildNotBuiltVoteValue() != null) {
+                return labelValue.getBuildNotBuiltVoteValue();
+            }
+            else {
+                return config.getVerdictCategory(labelName) != null ? config.getVerdictCategory(labelName).getDefaultBuildNotBuiltReportingValue() : null;
+            }
+        }
+        else {
+            //As bad as failure, for now
+            if (labelValue != null && labelValue.getBuildFailedVoteValue() != null) {
+                return labelValue.getBuildFailedVoteValue();
+            }
+            else {
+                return config.getVerdictCategory(labelName) != null ? config.getVerdictCategory(labelName).getDefaultBuildSuccessfulReportingValue() : null;
+            }
+        }
+    }
+
+    /**
+     * Returns the minimum verified value for the build results in the memory.
+     * If no builds have contributed to verified value, this method returns null
+     *
+     * @param memoryImprint the memory.
+     * @param onlyBuilt     only count builds that completed (no NOT_BUILT builds)
+     * @param labelName     the label for which you want to return the vote value
+     * @return the lowest verified value.
+     */
+    @CheckForNull
+    public Integer getMinimumLabelVoteValue(MemoryImprint memoryImprint, boolean onlyBuilt, String labelName) {
+        Integer labelVote = Integer.MAX_VALUE;
+        for (Entry entry : memoryImprint.getEntries()) {
+            if (entry == null) {
+                continue;
+            }
+            Run build = entry.getBuild();
+            if (build == null) {
+                continue;
+            }
+            Result result = build.getResult();
+            if (onlyBuilt && result == Result.NOT_BUILT) {
+                continue;
+            }
+
+            GerritTrigger trigger = GerritTrigger.getTrigger(entry.getProject());
+            if (shouldSkip(trigger.getSkipVote(), result)) {
+                continue;
+            }
+            Integer labelObj = getLabelVoteValue(result, trigger, labelName);
+            if (labelObj != null) {
+                labelVote = Math.min(labelVote, labelObj);
+            }
+        }
+
+        if (labelVote == Integer.MAX_VALUE) {
+            return null;
+        }
+
+        return labelVote;
+    }
+
+    /**
+     * Finds the code review value for the specified build result on the configured trigger.
+     *
+     * @param res     the build result.
      * @param trigger the trigger that might have overridden values.
      * @return the value.
      */
+    @Deprecated
```

Comment:
protected methods are not part of the API. If it is not used, it can be removed.

Reply:
They are still used in other public methods :(